src/basicio.cpp

```diff
@@ -953,11 +953,12 @@ namespace Exiv2
         }
 
         Impl::StructStat buf;
-        int ret = p_->stat(buf);
+        const int ret = p_->stat(buf);
 
         if (ret != 0)
-            return -1;
-        return buf.st_size;
+            return std::numeric_limits<size_t>::max();
```

Comment:
This should be actually equivalent as unsigned overflow is well defined ;-)

Reply:
Yes, I just changed it because the clang-tidy tool integrated in QtCreator was complaining there ;).